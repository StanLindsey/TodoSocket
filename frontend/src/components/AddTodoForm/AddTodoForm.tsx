import React, { useState } from "react";
import {
  makeStyles,
  Theme,
  createStyles,
  Paper,
  IconButton,
  InputBase
} from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      padding: "2px 4px",
      display: "flex",
      alignItems: "center"
    },
    input: {
      marginLeft: theme.spacing(1),
      flex: 1
    },
    iconButton: {
      padding: 10
    },
    divider: {
      height: 28,
      margin: 4
    }
  })
);

const AddTodoForm = ({ onSubmit }: { onSubmit: (todo: string) => void }) => {
  const classes = useStyles();
  const [text, setText] = useState("");

  const handleSubmit = (e: any) => {
    e.preventDefault();
    onSubmit(text);
    setText("");
  };

  return (
    <Paper
      component="form"
      className={classes.root}
      onSubmit={e => handleSubmit(e)}
    >
      <InputBase
        className={classes.input}
        placeholder="Add new Task"
        inputProps={{ "aria-label": "Add new task" }}
        onChange={e => setText(e.target.value)}
        value={text}
      />
      <IconButton
        type="submit"
        className={classes.iconButton}
        aria-label="submit new task"
      >
        <AddIcon />
      </IconButton>
    </Paper>
  );
};

export default AddTodoForm;
