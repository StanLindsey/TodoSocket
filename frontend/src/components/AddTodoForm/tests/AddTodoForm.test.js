import React from "react";
import renderer from "react-test-renderer";
import AddTodoForm from "..";

describe("app", () => {
  it("matches default snapshot", () => {
    const tree = renderer.create(<AddTodoForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
