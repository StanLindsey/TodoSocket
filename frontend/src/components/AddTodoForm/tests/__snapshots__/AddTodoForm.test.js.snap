// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app matches default snapshot 1`] = `
<form
  className="MuiPaper-root MuiPaper-elevation1 makeStyles-root-1 MuiPaper-rounded"
>
  <div
    className="MuiInputBase-root makeStyles-input-2"
    onClick={[Function]}
  >
    <input
      aria-label="Add new task"
      className="MuiInputBase-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Add new Task"
      type="text"
    />
  </div>
  <button
    aria-label="submit new task"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-3"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </span>
  </button>
</form>
`;
